app/controllers/users/totp_setup_controller.rb

```diff
@@ -1,6 +1,10 @@
 module Users
   class TotpSetupController < ApplicationController
-    before_action :confirm_two_factor_authenticated
+    # Because we allow initial auth through an app,
+    # so a user is not required to have a phone, we are
+    # not using the normal confirm_two_factor_authenticated method
+    # and instead are only checking that they are signed in
+    before_action :confirm_sign_in
```

Comment:
This opens up a security hole that allows anyone to bypass 2FA on someone else's account. Now, I just need to know your email and password, and I can then go to the authenticator_setup endpoint and configure my device and now I can sign into your account using the authenticator app on my device.

Reply:
OK, so how do we resolve that with the understanding that not everyone has a working phone (the reason behind this story)?

Reply:
One option might be to require full authentication (username/password/2fa) if they have any 2fa method set up.

Reply:
Although someone who abandoned the account creation process before setting up 2FA would be vulnerable as well, so that might not be the best solution. I'll have to think about this some more.

Reply:
Scratch that thought. That's how the app is already designed. If you haven't set up 2FA yet, someone else can set it up with their own phone number. Adding the TOTP option doesn't make it any less secure than it is now.

Reply:
Regardless of the 2fa method to be added, an account with just a username/password would be open to someone else adding a 2fa method if one wasn't already added.

Reply:
So just to confirm, after thinking about it more, we're feeling that it opens no new security vulnerability, correct?

Reply:
No, as is, there is still a new vulnerability that has been introduced. To fix it, we will need to add a check to make sure that the only way 2FA can be bypassed in order to reach the `authenticator_setup` endpoint is if the user has not set up 2FA yet.